src/Components/RelatedCollectionsRail/RelatedCollectionsRail.tsx

```diff
@@ -114,6 +118,19 @@ export const RelatedCollectionsRailFragmentContainer = createFragmentContainer(
       fragment RelatedCollectionsRail_collections on MarketingCollection
         @relay(plural: true) {
         ...RelatedCollectionEntity_collection
+        # We need this so we can filter out collections w/o artworks that would
+        # otherwise break the carousel.
+        artworksConnection(
+          first: 3
+          aggregations: [TOTAL]
+          sort: "-decayed_merch"
+        ) {
+          edges {
+            node {
+              id
+            }
+          }
+        }
```

Comment:
This `artworksConnection` field has to be hoisted up here bc of Relay data masking, is that right?

Reply:
Yea, we have this pattern in a few places. The exact same snippet, copied into here, will ensure we don't kick off redundant data requests (differing slightly in some query params maybe), when the goal is to have an exist-y check on the data.

And then yea, the child component has the actual data fragment, but a parent component often needs to know whether a child component will render or not, for a layout reason.

Short of larger refactors with the component heirarchy, I don't know a great solution! This might be a reason to actually use the `@relay(mask: false)` directive, or copy the whole fragment into a string that can be shared and interpolated here (to help avoid a bug of updating one of the two copies).

Reply:
I've been encouraged to avoid relay masking wherever possible-- in this case the duplication means that if we no longer need this functionality, it's very clear to just remove it from this one spot!